src/plugins/opencl_support/src/OpenCLSupportSettingsController.cpp

```diff
@@ -19,125 +19,155 @@
+        QLabel *gpusDiscoveredLabel = new QLabel(actualText, this);
         gpusDiscoveredLabel->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Preferred);
 
-        vLayout->addWidget( gpusDiscoveredLabel );
+        vLayout->addWidget(gpusDiscoveredLabel);
 
-        foreach( OpenCLGpuModel * m, gpus ) {
-            vLayout->setAlignment( Qt::AlignLeft | Qt::AlignTop );
-            QHBoxLayout * hLayout = new QHBoxLayout(this);
+        foreach (OpenCLGpuModel *m, gpus) {
+            vLayout->setAlignment(Qt::AlignLeft | Qt::AlignTop);
+            QHBoxLayout *hLayout = new QHBoxLayout(this);
 
-            QString gpuText = m->getName() + " " + QString::number(m->getGlobalMemorySizeBytes() / (1024*1024)) + " Mb";
-            QCheckBox * check = new QCheckBox( gpuText, this );
+            QString gpuText = m->getName() + " " + QString::number(m->getGlobalMemorySizeBytes() / (1024 * 1024)) + " Mb";
+            QRadioButton *rb = new QRadioButton(gpuText, this);
+            rb->setChecked(m->isEnabled());
 
-            check->setChecked(true);
-
-            gpuEnableChecks.push_back(check);
-            hLayout->addWidget(check);
-            vLayout->addLayout( hLayout );
+            gpuRadioButtons.push_back(rb);
+            hLayout->addWidget(rb);
+            vLayout->addLayout(hLayout);
         }
         setLayout(vLayout);
     }
-
 }
 
-void OpenCLSupportSettingsPageWidget::setState( AppSettingsGUIPageState * _state ) {
-    OpenCLSupportSettingsPageState * state = qobject_cast<OpenCLSupportSettingsPageState*>(_state);
-    assert( state->enabledGpus.size() == gpuEnableChecks.size() );
-
-    for( int i = 0, end = state->enabledGpus.size(); i < end; ++i ) {
-        gpuEnableChecks.at(i)->setChecked( state->enabledGpus.at(i) );
+void OpenCLSupportSettingsPageWidget::setState(AppSettingsGUIPageState *_state) {
+    CHECK(!gpuRadioButtons.isEmpty(), )
+
+    OpenCLSupportSettingsPageState *state = qobject_cast<OpenCLSupportSettingsPageState *>(_state);
+    SAFE_POINT(nullptr != state, "OpenCLSupportSettingsPageState isn't found", );
+    
+    const QString enbledGpuName = state->getEnabledGpuName();
+    bool checkedRadioButtonwasFound = false;
+    foreach(QRadioButton * rb, gpuRadioButtons) {
+        if (rb->text().startsWith(enbledGpuName)) {
+            rb->setChecked(true);
+            checkedRadioButtonwasFound = true;
+        } else {
+            rb->setChecked(false);
+        }
+    }
+    if (!checkedRadioButtonwasFound) {
+        gpuRadioButtons.first()->setChecked(true);
     }
 }
 
-AppSettingsGUIPageState* OpenCLSupportSettingsPageWidget::getState( QString & /*err*/ ) const {
-    OpenCLSupportSettingsPageState * state = new OpenCLSupportSettingsPageState( gpuEnableChecks.size() );
+AppSettingsGUIPageState *OpenCLSupportSettingsPageWidget::getState(QString & /*err*/) const {
+    CHECK(!gpuRadioButtons.isEmpty(), nullptr);
```

Comment:
Do not return nullptr, the caller doesn't expect it. Return a state with an empty name.